app/main.py

```diff
@@ -1,34 +1,94 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(
+        self,
+        row: int,
+        column: int,
+        is_alive: bool = True
+    ) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+        self,
+        start: tuple,
+        end: tuple,
+        is_drowned: bool = False
+    ) -> None:
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
+        self.decks = []
+        st = self.start[0]
+        fix_pos = self.end[1]
+        iter_pos = self.end[0]
+        row = True
+        if self.start[0] == self.end[0]:
+            row = False
+            fix_pos = self.end[0]
+            iter_pos = self.end[1]
+            st = self.start[1]
+        for deck in range(st, iter_pos + 1):
+            if not row:
+                self.decks.append(Deck(fix_pos, deck))
+            else:
+                self.decks.append(Deck(deck, fix_pos))
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def get_deck(
+        self,
+        row: int,
+        column: int
+    ) -> Deck:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def fire(
+        self,
+        row: int,
+        column: int
+    ) -> None:
+        deck = self.get_deck(row, column)
+        deck.is_alive = False
+        ship_status = [cell.is_alive for cell in self.decks]
+        if not any(ship_status):
+            self.is_drowned = True
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(self, ships: list) -> None:
+        self.ships = ships
+        self.field = [[None for _ in range(10)] for _ in range(10)]
+        self.ships = [Ship(value[0], value[1]) for value in self.ships]
+        for ship in self.ships:
+            for cell in ship.decks:
+                self.field[cell.row][cell.column] = ship
+
+    def fire(self, location: tuple) -> str:
+        field_loc = self.field[location[0]][location[1]]
```

Comment:
```suggestion
        ship = self.field[location[0]][location[1]]
```
Rename this variable